libpldmresponder/fru.cpp

```diff
@@ -453,6 +454,43 @@ void FruImpl::removeIndividualFRU(const std::string& fruObjPath)
     } // sm00 this can be RECORDS_DELETED also for adapter pdrs
 }
 
+void FruImpl::deleteFruRecord(uint16_t rsi)
+{
+    std::vector<uint8_t> updatedFruTbl;
+    const struct pldm_fru_record_data_format* recordSetSrc =
+        reinterpret_cast<const struct pldm_fru_record_data_format*>(
+            table.data());
+
+    const struct pldm_fru_record_tlv* tlv;
+    size_t pos = 0;
+
+    while (table.size() > pos)
+    {
+        size_t recordLen = sizeof(struct pldm_fru_record_data_format) -
+                           sizeof(struct pldm_fru_record_tlv);
+        tlv = recordSetSrc->tlvs;
+
+        for (int i = 0; i < recordSetSrc->num_fru_fields; i++)
```

Comment:
Please use unsingned int (uint) instead of signed one

Reply:
Done